src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -255,6 +255,22 @@
   [cert-or-csr :- CertOrCSR]
   (:dns-name (get-extension-value cert-or-csr subject-alt-name-oid)))
 
+(schema/defn ^:always-validate subject-ip-alt-names :- SSLExtension
+  "Create a Subject Alternative Names extensions (OID=2.5.29.17) which contains
+  a list of DNS names as alternative names. The `critical` argument sets the
+  criticality flag of this extension."
+  [alt-names-list :- [schema/Str]
+   critical :- Object]
+  {:oid      subject-alt-name-oid
+   :critical (boolean critical)
+   :value    {:ip alt-names-list}})
+
+(schema/defn ^:always-validate get-subject-ip-alt-names :- (schema/maybe [schema/Str])
+ "Given a certificate or CSR, return the list of IP alternative names on the
+  Subject Alternative Names extension, or nil if the extension is not present."
+ [cert-or-csr :- CertOrCSR]
+ (:ip (get-extension-value cert-or-csr subject-alt-name-oid)))
```

Comment:
Should this be combined with the DNS version and made generic (like not referring to IP or DNS)? Given the defaults that we add, I think we would pretty much always need an extension that has a mix.

Reply:
In the puppetserver code I need to preface all the IP SANS with "IP:" and the dns ones with "DNS:", so that's why I was thinking of them separately.

Reply:
^ I may end up adding a combined method as well if needed (as the ticket progresses).